packages/client/src/webAPI/webAudioAPI.ts

```diff
@@ -0,0 +1,48 @@
+export default class AudioAPI {
+  private static _instance: AudioAPI
+  private _audioContext!: AudioContext
+  private _audioRef!: HTMLMediaElement
+  private _source!: MediaElementAudioSourceNode | null
+  private _gainNode!: GainNode | null
+
+  constructor() {
+    if (AudioAPI._instance) {
+      return AudioAPI._instance
+    }
+
+    this._audioContext = new AudioContext()
+    this._source = null
+    this._gainNode = null
+    AudioAPI._instance = this
+  }
+
+  public init(audioRef: HTMLMediaElement) {
+    if (this._source == null) {
```

Comment:
Небезопасно проверять двойным равной, лучше всегда проверять тройной
```suggestion
    if (this._source === null) {
```

Reply:
[Сравнение с использованием ==](https://developer.mozilla.org/ru/docs/Web/JavaScript/Equality_comparisons_and_sameness#%D1%81%D1%80%D0%B0%D0%B2%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5_%D1%81_%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5%D0%BChttps://developer.mozilla.org/ru/docs/Web/JavaScript/Equality_comparisons_and_sameness#%D1%81%D1%80%D0%B0%D0%B2%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5_%D1%81_%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5%D0%BC)